src/components/List/ListItem/ListItem.js

```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import './ListItem.css';
+import vectorLayer, {view, defaultStyle, hoverStyle, selectStyle} from '../../../map.js'
+
+const ListItem = (props) => (
+    <tr className="list-items" 
+    onMouseOver={() => {vectorLayer.getSource().uidIndex_[props.id].setStyle(hoverStyle)}} 
+    onMouseLeave={() => {vectorLayer.getSource().uidIndex_[props.id].setStyle(defaultStyle)}}
+    onClick={() => {vectorLayer.getSource().uidIndex_[props.id].setStyle(selectStyle);view.animate({center: [vectorLayer.getSource().uidIndex_[props.id].geometryChangeKey_.target.extent_[0],vectorLayer.getSource().uidIndex_[props.id].geometryChangeKey_.target.extent_[1]],duration: 1000,zoom: 18});}} >
```

Comment:
podajesz do onclicka bardzo złożoną funkcję, można ją spokojnie wyodrębnić jako funkcja wewnątrz komponentu funkcyjnego - robimy to podobnie jak definicję komponentu funkcyjnego tylko w jego wnętrzu. 

```
const Component = (props) => {
  const fn = () => {}

  return (
   <button onClick={fn} />
  )
}
```